user_tools/src/spark_rapids_tools/cmdli/argprocessor.py

```diff
@@ -265,6 +265,25 @@ class ToolUserArgModel(AbsToolUserArgModel):
     This is used as doing preliminary validation against some of the common pattern
     """
     eventlogs: Optional[str] = None
+    jvm_heap_size: Optional[int] = None
+
+    def is_concurrent_submission(self):
+        return False
+
+    def process_jvm_args(self):
+        # JDK8 uses parallel-GC by default. Set the GC algorithm to G1GC
+        self.p_args['toolArgs']['jvmGC'] = '+UseG1GC'
+        jvm_heap = self.jvm_heap_size
+        if jvm_heap is None:
+            # set default GC heap size based on the virtual memory of the host.
+            jvm_heap = Utilities.get_system_memory_in_gb()
```

Comment:
Should we upper bound `jvm_heap` by `Utilities.get_system_memory_in_gb()` in any case?

Reply:
I am not a big fan of uppering the maximum heap size because this will eat the host machine.
Leaving 20% for the system is safer.
We enforce a minimum of 8 GB heap even when the host does not have that much memory but I think that's fine because processing batches should not be done on a host with that small memory anyway.

Reply:
By upper bounding I wanted to prevent the case when user mistakenly sets heap memory to a very large value?

Reply:
We can ignore this comment. Larger heap values could be beneficial for us.